src/math/TransformBoundedField.cc

```diff
@@ -0,0 +1,172 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2008-2014 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <memory>
+#include <string>
+
+#include "ndarray/eigen.h"
+#include "astshim.h"
+#include "lsst/afw/math/TransformBoundedField.h"
+#include "lsst/afw/table/io/InputArchive.h"
+#include "lsst/afw/table/io/OutputArchive.h"
+#include "lsst/afw/table/io/CatalogVector.h"
+#include "lsst/afw/table/aggregates.h"
+
+namespace lsst {
+namespace afw {
+namespace math {
+
+TransformBoundedField::TransformBoundedField(
+        geom::Box2I const& bbox,
+        geom::Transform<geom::Point2Endpoint, geom::GenericEndpoint> const& transform)
+        : BoundedField(bbox), _transform(transform) {
+    if (transform.getToEndpoint().getNAxes() != 1) {
+        throw LSST_EXCEPT(pex::exceptions::InvalidParameterError,
+                          "To EndPoint of transform must have 1 axis but has " +
+                                  std::to_string(transform.getToEndpoint().getNAxes()));
+    }
+}
+
+double TransformBoundedField::evaluate(geom::Point2D const& position) const {
+    return _transform.applyForward(position)[0];
```

Comment:
_transform is of a type `Transform<Point2Endpoint, GenericEndpoint>` and the generic endpoint has a single axis. As such, the single-point version of `applyForward` takes a Point2D and returns a vector with one element (for the one axis). So from that perspective it is certainly safe. Does that answer your question, or were you thinking of something else?